packages/e2e-tests/specs/editor/blocks/cover.test.js

```diff
@@ -126,24 +126,20 @@ describe( 'Cover', () => {
 			'.components-circular-option-picker__option-wrapper:first-child button'
 		);
 
-		// Select the cover block.By default the child paragraph gets selected.
+		// Select the cover block. By default the child paragraph gets selected.
 		await page.click(
 			'.edit-post-header-toolbar__document-overview-toggle'
 		);
 		await page.click(
 			'.block-editor-list-view-block__contents-container a'
 		);
 
-		const heightInput = (
-			await page.$x(
-				'//div[./label[contains(text(),"Minimum height of cover")]]/following-sibling::div//input'
-			)
-		 )[ 0 ];
+		const heightInput = await page.waitForSelector(
+			'input[id*="block-cover-height-input"]'
+		);
```

Comment:
Is there an idiomatic way in Puppeter/Playwright to find an input labelled as "Minimum height of cover"? React Testing Library can do it with the `.getByLabelText` query. If there is a `<label for="x">` with that text and the `for` attribute points to an `<input id="x">`, it will return the `input`.

Reply:
> Is there an idiomatic way in Puppeter/Playwright to find an input labelled as "Minimum height of cover"?

Not in Puppeteer, but fortunately, [there is one in Playwright](https://playwright.dev/docs/api/class-framelocator#frame-locator-get-by-label):

```js
await page.getByLabel("Minimum height of cover");
```

We'll definitely use it when this spec is migrated.